strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/providers/repository/HostedRepositoryProvider.java

```diff
@@ -157,28 +158,26 @@ public RepositoryPath resolvePath(String storageId,
         Repository repository = getConfiguration().getStorage(storageId).getRepository(repositoryId);
         if (repository == null)
         {
-            logger.warn(String.format("Trying to resolve repository which not exists [%s].", repositoryId));
+            logger.error(String.format("Trying to resolve repository which not exists [%s].", repositoryId));
             return null;
         }
         
-        final LayoutProvider layoutProvider = layoutProviderRegistry.getProvider(repository.getLayout());
+        LayoutProvider layoutProvider = layoutProviderRegistry.getProvider(repository.getLayout());
         if (layoutProvider == null)
         {
-            logger.warn(String.format("Trying to resolve repository with unknown layout [%s].", repository.getLayout()));
+            logger.error(String.format("Trying to resolve repository with unknown layout [%s].", repository.getLayout()));
             return null;
         }
         
-        final RepositoryPath artifactPath = layoutProvider.resolve(repository).resolve(path);
-
+        RepositoryPath artifactPath = layoutProvider.resolve(repository).resolve(path);
         logger.debug(" -> Checking local cache for {} ...", artifactPath);
-        if (layoutProvider.containsPath(repository, path))
+        if (!Files.exists(artifactPath))
         {
-            logger.debug("The artifact {} was found in the local cache", artifactPath);
-            return artifactPath;
+            logger.debug("The artifact {} was not found in the local cache", artifactPath);
```

Comment:
Cosmetic: Could you please leave empty lines around logging blocks? For example
```
String s = "blah";
logger.info(s);
logger.info("something is going on here");
doSomethingWith(s);
```
Should be:
```
String s = "blah";

logger.info(s);
logger.info("something is going on here");

doSomethingWith(s);
```